src/legacy/server/saved_objects/service/lib/errors.ts

```diff
@@ -30,112 +39,113 @@ function decorate(error, errorCode, statusCode, message) {
     statusCode,
     message,
     override: false,
-  });
+  }) as DecoratedError;
 
   boom[code] = errorCode;
 
   return boom;
 }
 
-export function isSavedObjectsClientError(error) {
-  return error && !!error[code];
+export function isSavedObjectsClientError(error: any): error is DecoratedError {
+  return Boolean(error && error[code]);
 }
 
 // 400 - badRequest
 const CODE_BAD_REQUEST = 'SavedObjectsClient/badRequest';
-export function decorateBadRequestError(error, reason) {
+export function decorateBadRequestError(error: Error, reason?: string) {
   return decorate(error, CODE_BAD_REQUEST, 400, reason);
 }
-export function createBadRequestError(reason) {
+export function createBadRequestError(reason?: string) {
   return decorateBadRequestError(new Error('Bad Request'), reason);
 }
-export function createUnsupportedTypeError(type) {
+export function createUnsupportedTypeError(type: string) {
   return createBadRequestError(`Unsupported saved object type: '${type}'`);
 }
-export function isBadRequestError(error) {
-  return error && error[code] === CODE_BAD_REQUEST;
+export function isBadRequestError(error: Error | DecoratedError) {
+  return isSavedObjectsClientError(error) && error[code] === CODE_BAD_REQUEST;
 }
 
 // 400 - invalid version
 const CODE_INVALID_VERSION = 'SavedObjectsClient/invalidVersion';
-export function createInvalidVersionError(versionInput) {
+export function createInvalidVersionError(versionInput?: string) {
   return decorate(Boom.badRequest(`Invalid version [${versionInput}]`), CODE_INVALID_VERSION, 400);
 }
-export function isInvalidVersionError(error) {
-  return error && error[code] === CODE_INVALID_VERSION;
+export function isInvalidVersionError(error: Error | DecoratedError) {
+  return isSavedObjectsClientError(error) && error[code] === CODE_INVALID_VERSION;
 }
 
 // 401 - Not Authorized
 const CODE_NOT_AUTHORIZED = 'SavedObjectsClient/notAuthorized';
-export function decorateNotAuthorizedError(error, reason) {
+export function decorateNotAuthorizedError(error: Error, reason?: string) {
   return decorate(error, CODE_NOT_AUTHORIZED, 401, reason);
 }
-export function isNotAuthorizedError(error) {
-  return error && error[code] === CODE_NOT_AUTHORIZED;
+export function isNotAuthorizedError(error: Error | DecoratedError) {
+  return isSavedObjectsClientError(error) && error[code] === CODE_NOT_AUTHORIZED;
 }
 
 // 403 - Forbidden
 const CODE_FORBIDDEN = 'SavedObjectsClient/forbidden';
-export function decorateForbiddenError(error, reason) {
+export function decorateForbiddenError(error: Error, reason?: string) {
   return decorate(error, CODE_FORBIDDEN, 403, reason);
 }
-export function isForbiddenError(error) {
```

Comment:
optional: inconsistency in signatures 
`function isNotAuthorizedError(error: Error | DecoratedError) ` 
but here
`function  isForbiddenError (error: any`

Reply:
😩